zookeeper-server/src/main/java/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -311,13 +343,19 @@ public void processRequest(Request request) {
                     ((QuorumZooKeeperServer) zks).self.getQuorumVerifier().toString().getBytes(),
                     rc.stat);
                 err = Code.get(rc.err);
+                addAuditLog(request, cnxn, AuditConstants.OP_RECONFIG, rc.path, null, null, err);
                 break;
             }
             case OpCode.setACL: {
                 lastOp = "SETA";
                 rsp = new SetACLResponse(rc.stat);
                 err = Code.get(rc.err);
                 requestPathMetricsCollector.registerRequest(request.type, rc.path);
+                /** Here audit enable check is done to avoid getACLs() call in case audit is disabled. */
+                if (ZKAuditLogger.isAuditEnabled()) {
```

Comment:
Similarly pass the `request` object to `addAuditLog()` and extract the required information inside the `if`.

Reply:
Refactored complete code and this is addressed in code refactoring.